lib/state_machine.rb

```diff
@@ -1,7 +1,162 @@
 # frozen_string_literal: true
 
 module StateMachine
-  def test
-    'test'
+  InitialStateDuplicateError = Class.new(RuntimeError)
+  StateAlreadyDefinedError = Class.new(RuntimeError)
+  InvalidStateError = Class.new(RuntimeError)
+  InvalidTransitionError = Class.new(RuntimeError)
+  GuardIsNotBooleanError = Class.new(RuntimeError)
```

Comment:
better to have some base class to inherit from, `StateMachneError`, `BaseError` and better inherit base class from `StandardError`